### PR TITLE
[Doc] update doc page for distributed partition pipeline

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: docs/source/guide/distributed-preprocessing.rst [hunk 1]
```diff
@@ -4,9 +4,18 @@
 ------------------------------------------
 
 Before launching training jobs, DGL requires the input data to be partitioned
-and distributed to the target machines. For relatively small graphs, DGL
-provides a partitioning API :func:`~dgl.distributed.partition_graph` that
-partitions an in-memory :class:`~dgl.DGLGraph` object. It supports
+and distributed to the target machines. In order to handle diffierent scales
```
Comment (jermainewang) [inline docs/source/guide/distributed-preprocessing.rst:7]: diffierent -> different

File: docs/source/guide/distributed-preprocessing.rst [hunk 2]
```diff
@@ -4,9 +4,18 @@
...
+of graphs, DGL provides 2 partitioning approaches:
+
+* A partitioning API for relatively small graphs.
```
Comment (jermainewang) [inline docs/source/guide/distributed-preprocessing.rst:10]: `* A partitioning API for graphs that can fit in a single machine memory.`

File: docs/source/guide/distributed-preprocessing.rst [hunk 3]
```diff
@@ -4,9 +4,18 @@
...
+* A distributed partition pipeline for relatively large graphs.
```
Comment (jermainewang) [inline docs/source/guide/distributed-preprocessing.rst:11]: `* A distributed partition pipeline for graphs beyond a single machine capacity.`

File: docs/source/guide/distributed-preprocessing.rst [hunk 4]
```diff
@@ -361,8 +424,82 @@ An example IP configuration file is as follows:
...
+
 During data dispatching, DGL assumes that the combined CPU RAM of the cluster
-is able to hold the entire graph data. Moreover, the number of machines (IPs) must be the
-same as the number of partitions. Node ownership is determined by the result
+is able to hold the entire graph data. Node ownership is determined by the result
 of partitioning algorithm where as for edges the owner of the destination node
 also owns the edge as well.
+
+Load partitioned graphs
+^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (jermainewang) [inline docs/source/guide/distributed-preprocessing.rst:475]: Maybe move this section to after "ID mapping"?

File: docs/source/guide/distributed-preprocessing.rst [hunk 5]
```diff
@@ -4,9 +4,18 @@
...
-and distributed to the target machines. For relatively small graphs, DGL
-provides a partitioning API :func:`~dgl.distributed.partition_graph` that
-partitions an in-memory :class:`~dgl.DGLGraph` object. It supports
+and distributed to the target machines. In order to handle diffierent scales
+of graphs, DGL provides 2 partitioning approaches:
+
+* A partitioning API for relatively small graphs.
+* A distributed partition pipeline for relatively large graphs.
+
+Partitioning API
```
Comment (jermainewang) [inline docs/source/guide/distributed-preprocessing.rst:13]: Change to "7.1.1 Partitioning API"

File: docs/source/guide/distributed-preprocessing.rst [hunk 6]
```diff
@@ -49,6 +58,61 @@ CPU RAM to hold the entire graph structure and features, which may not be viable
...
+
+    node_map, edge_map = dgl.distributed.partition_graph(g, 'graph_name', 4, '/tmp/test',
+                                                         balance_ntypes=g.ndata['train_mask'],
+                                                         return_mapping=True)
+    # Let's assume that node_emb is saved from the distributed training.
+    orig_node_emb = th.zeros(node_emb.shape, dtype=node_emb.dtype)
+    orig_node_emb[node_map] = node_emb
+
+
 Parallel Data Preparation Pipeline
```
Comment (jermainewang) [inline docs/source/guide/distributed-preprocessing.rst:116]: Change to "7.1.2 Distributed Graph Partitioning Pipeline"

File: docs/source/guide/distributed-preprocessing.rst (hunk 4 above)
Comment (Rhett-Ying) [reply docs/source/guide/distributed-preprocessing.rst:475]: ok

**Review by jermainewang — APPROVED** [review]

**Pull request merged**